### PR TITLE
python3Packages.posthog: fix package testing

### PR DESCRIPTION
**Pull request by blaggacao**
###### Description of changes

Official PostHog python library

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blaggacao** [comment]
Result of `nixpkgs-review pr 242575` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.posthog</li>
    <li>python311Packages.posthog</li>
  </ul>
</details>

File: pkgs/development/python-modules/posthog/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/posthog/default.nix:16]:
```suggestion
```

File: pkgs/development/python-modules/posthog/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/posthog/default.nix:45]:
```suggestion
```
Included in hook

File: pkgs/development/python-modules/posthog/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/posthog/default.nix:42]:
```suggestion
```
We don't want to do linting

File: pkgs/development/python-modules/posthog/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/posthog/default.nix:44]:
```suggestion
```
Or coverage collecting

File: pkgs/development/python-modules/posthog/default.nix
Comment (SuperSandro2000) [inline pkgs/development/python-modules/posthog/default.nix:43]:
```suggestion
```

**Comment by blaggacao** [comment]
cc @happysalada

**Comment by happysalada** [comment]
Thanks! This looks good.

On a style note, I personally like the comments that tell you the built inputs and check inputs. if there is an update, it makes it easier to follow the setup.py and know which requirements are which.
But I understand this is a matter of style, so I'm good either way.

I'll wait for ofborg to finish before merging. (unless you need/want to do more work on this).

**Comment by blaggacao** [comment]
> tell you the built inputs and check inputs

[Agreed](https://github.com/NixOS/nixpkgs/compare/2b581781fc4406f5920a6afe7b247c07dfa444ac..b406d7e742ce124bd3921fcff7bd1299bd8a1411), this gets us one step closer to normalized internal interfaces (i.e. testing in separate drv in the far distant future).

**Comment by happysalada** [comment]
Thanks! Let's wait for ofborg to finish and merge this.

**Review by happysalada — APPROVED** [review]

**Pull request merged**
